File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        int m = n;
        int temp = 0;
        if(m==0){
            return 1;
        }
        while(m!=0){
            temp = (temp<<1)|1;
            m = m>>1;
        }
        int ans = (~n)&temp;

        return ans;
        
    }
};

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if(n==0) return 1;
        int number = n;
        int mask = 0;
        while( number != 0 ){
            number = number >> 1;
            mask = mask << 1 | 1;
        }
        
        return (~n)&mask;
        
    }
}